lib/modules/datasource/metadata.ts

```diff
@@ -233,3 +238,13 @@ export function shouldDeleteHomepage(
   }
   return massagedSourceUrl === homepage;
 }
+
+function tryParseGitlabUrl(url: string): string | null {
+  const result = parseGitlabUrl(url);
+
+  if (result === null || result.type !== 'gitlab') {
+    return null;
+  }
+
+  return `https://gitlab.com/${result.owner}/${result.name}`;
```

Comment:
this seems also wrong. what if parsed tape is gitlab but different host (selfhosted)?

Reply:
I think right nowthe library only works for certain official domains. This function only tries to mitigate the issue discussed in https://github.com/renovatebot/renovate/discussions/26721 without introducing any additional/changed behaviour :)
Thats also why `parse()` is still used as the front runner.

Reply:
@otbe how wide is this intended to be? e.g. solving for gitlab.com only?